homeassistant/components/toon/config_flow.py

```diff
@@ -93,6 +94,7 @@ async def _create_entry(self, agreement: Agreement) -> FlowResult:
         await self.async_set_unique_id(agreement.agreement_id)
         self._abort_if_unique_id_configured()
 
+        assert self.data is not None
```

Comment:
Instead adjust the typing of `self.data`, which cannot/should not be `None`.

Reply:
Done